src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientMetricDiagnosticListener.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Reflection;
+
+namespace OpenTelemetry.Instrumentation.SqlClient.Implementation;
+
+#if NET6_0_OR_GREATER
+internal class SqlClientMetricDiagnosticListener : ListenerHandler
```

Comment:
Since both this new `SqlClientMetricDiagnosticListener` and the existing `SqlClientDiagnosticListener` rely on the same events, can they really work well together?

E.g. if I change the order in the `SqlClientCallsAreCollectedSuccessfully` and first call `meterProviderBuilder.AddSqlClientInstrumentation();` and after that `traceProviderBuilder.AddSqlClientInstrumentation` (so first this new class is registered and then the existing `SqlClientDiagnosticListener`), then the test fail, because `SqlClientDiagnosticListener.SqlMicrosoftAfterExecuteCommand` never fires in this class. 

I believe users can register span and metric creation in any order and nothing would prevent them from doing so, right? If that's the case I think this design should be reconsidered.

I totally agree in general with the separation of the class that takes care of the activity and this one that takes care of metrics, but I think in this form, it doesn't really work.

Maybe an option would be to keep a single class handling the events, but then the activity creation and the metric creation could still be in 2 separate classes and the one handling the events can just call both.